src/Tizen.MachineLearning.Inference/Tizen.MachineLearning.Inference/Pipeline.cs

```diff
@@ -0,0 +1,498 @@
+            sce.NewDataCreated -= newDataCreated;
+        }
+
+        /// <summary>
+        /// Adds an input data frame to source node.
+        /// </summary>
+        /// <param name="srcNodeName">The name of source node</param>
+        /// <param name="data">TensorsData to be inputed</param>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when failed to input TensorsData because of internal error.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void InputData(string srcNodeName, TensorsData data)
+        {
+            NNStreamer.CheckNNStreamerSupport();
+
+            if (string.IsNullOrEmpty(srcNodeName))
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "Node Name is invalid");
+
+            if (data == null)
+                throw NNStreamerExceptionFactory.CreateException(NNStreamerError.InvalidParameter, "TensorsData is invalid");
+
+            NNStreamerError ret = NNStreamerError.None;
+
+            /* Get src node handle */
+            NodeHandleInfo hInfo;
+            if (_nodeHandleList.ContainsKey(srcNodeName))
+            {
+                hInfo = _nodeHandleList[srcNodeName];
+            }
+            else
+            {
+                IntPtr srcHandle = IntPtr.Zero;
+                ret = Interop.Pipeline.GetSrcHandle(_handle, srcNodeName, out srcHandle);
+                NNStreamer.CheckException(ret, "Failed to get Source Node handle");
+
+                hInfo = new NodeHandleInfo(srcHandle, HandleType.Source);
+                _nodeHandleList.Add(srcNodeName, hInfo);
+            }
+
+            /* Prepare TensorsData */
+            data.PrepareInvoke();
+
+            /* Input data */
+            ret = Interop.Pipeline.InputSrcData(hInfo.Handle, data.GetHandle(), PipelineBufferPolicy.NotFreed);
+            NNStreamer.CheckException(ret, "Failed to input tensors data to source node: " + srcNodeName);
+        }
+
+        /// <summary>
+        /// Controls the valve. Set the flag true to open, false to close.
+        /// </summary>
+        /// <param name="valveNodeName">The name of valve node</param>
+        /// <param name="open">The flag to control the flow (True for open)</param>
+        /// <feature>http://tizen.org/feature/machine_learning.inference</feature>
+        /// <exception cref="NotSupportedException">Thrown when the feature is not supported.</exception>
+        /// <exception cref="ArgumentException">Thrown when the method failed due to an invalid parameter.</exception>
+        /// <since_tizen> 8 </since_tizen>
+        public void ControlValve(string valveNodeName, bool open)
```

Comment:
What is `Vavle`? there is no description about it
and how about rename to `UpdateValve`?

Reply:
valve is a simple element to pass/stop the data flow in pipeline. (https://gstreamer.freedesktop.org/data/doc/gstreamer/head/gstreamer-plugins/html/gstreamer-plugins-valve.html)

I will update description like below.
```
Controls the valve. Set the flag true to open (let the flow pass), false to close (stop the flow).
```